azext_iot/assets/user_messages.py

```diff
@@ -16,7 +18,7 @@ def error_no_hub_or_login_on_input(entity_type="IoT Hub"):
             "Please provide an {0} entity name (via the '{1}' or '-n' parameter)"
             " or {0} connection string via --login..."
         ).format(entity_type, PARAM_MAPPING[entity_type])
-    return ("Please provide either a hub name or a DPS name.")
+    raise InvalidArgumentValueError("Please provide either a hub name or a DPS name.")
```

Comment:
_If_ this error _should only_ raise during development (I believe that was mentioned in a prior iteration PR), then it can be a [TypeError](https://docs.python.org/3/library/exceptions.html#TypeError) with msg something like "entity_type requires a str value". the `if entity_type:` condition can be modified to `if entity_type and isinstance(entity_type, str):`. Also would be good to add the type hint to entity_type,  `entity_type: str`.